lib/worker_base.js

```diff
@@ -43,7 +43,7 @@ class WorkerBase {
    */
   getLastProcessedPosition() {
     return {
-      blockNumber: this.lastExportedBlock,
+      blockNumber: this.lastBufferedBlock,
```

Comment:
That's wrong. This is the state we would preserve in ZK from which we would recover on restart. Last buffered block is too optimistic, this is not yet in Kafka.